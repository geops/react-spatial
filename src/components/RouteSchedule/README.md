#

This demonstrates the use of RouteSchedule.

```jsx
import React, { useState, useEffect } from 'react';
import { Layer, TrajservLayer } from 'mobility-toolbox-js/ol';
import Tile from 'ol/layer/Tile';
import OSM from 'ol/source/OSM';
import BasicMap from 'react-spatial/components/BasicMap';
import RouteSchedule from 'react-spatial/components/RouteSchedule';
import FilterButton from 'react-spatial/components/FilterButton';
import FollowButton from 'react-spatial/components/FollowButton';
import Filter from 'react-spatial/images/FilterButton/filter.svg';
import Follow from 'react-spatial/images/FollowButton/follow.svg';

// The `apiKey` used here is for demonstration purposes only.
// Please get your own api key at https://developer.geops.io/.
const trackerLayer = new TrajservLayer({
  apiKey: window.apiKey,
});

const layers = [
  new Layer({
    olLayer: new Tile({
      source: new OSM(),
    }),
  }),
  trackerLayer,
];

function RouteScheduleExample() {
  const [lineInfos, setLineInfos] = useState(null);
  const [filterActive, setFilterActive] = useState(false);
  const [followActive, setFollowActive] = useState(false);
  const [center, setCenter] = useState([951560, 6002550]);
  useEffect(()=> {
    trackerLayer.onClick((newLineInfos)=> {
      setLineInfos(newLineInfos);
    });
  }, []);

  useEffect(()=> {
    trackerLayer.map.updateSize();
  }, [lineInfos]);

  return (
    <div className="rt-route-schedule-example">
      <RouteSchedule
        lineInfos={lineInfos}
        trackerLayer={trackerLayer}
        renderHeaderButtons={routeIdentifier => (
          <>
            <FilterButton
              title="Filter"
              active={filterActive}
              onClick={active => setFilterActive(active)}
              routeIdentifier={routeIdentifier}
              trackerLayer={trackerLayer}
            >
              <Filter />
            </FilterButton>
            <FollowButton
              setCenter={setCenter}
              title="Follow"
              active={followActive}
              onClick={active => setFollowActive(active)}
              routeIdentifier={routeIdentifier}
              trackerLayer={trackerLayer}
            >
              <Follow />
            </FollowButton>
          </>
        )}
      />
      <BasicMap
        center={center}
        zoom={15}
        layers={layers}
        tabIndex={0}
      />
    </div>
  );
}

<RouteScheduleExample />;
```
