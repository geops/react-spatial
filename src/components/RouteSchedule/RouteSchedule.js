/* eslint-disable react/no-unused-prop-types */
/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { TrackerLayer } from 'mobility-toolbox-js/ol';
import { bgColors } from 'mobility-toolbox-js/common/trackerConfig';
import {
  getHoursAndMinutes,
  getDelayString,
} from 'mobility-toolbox-js/common/timeUtils';
import ReactTransitPropTypes from '../../propTypes';
import firstStation from '../../images/RouteSchedule/firstStation.png';
import station from '../../images/RouteSchedule/station.png';
import lastStation from '../../images/RouteSchedule/lastStation.png';

/**
 * Returns a color class to display the delay.
 * @param {Number} time Delay time in milliseconds.
 */
const getDelayColor = (time) => {
  const secs = Math.round(((time / 1800 / 2) * 3600) / 1000);
  if (secs >= 3600) {
    return 'dark-red';
  }
  if (secs >= 500) {
    return 'middle-red';
  }
  if (secs >= 300) {
    return 'light-red';
  }
  if (secs >= 180) {
    return 'orange';
  }
  return 'green';
};

/**
 * Returns true if the train doesn't stop to the station.
 * @param {Object} stop Station information.
 */
const isNotStop = (stop) => {
  return !stop.arrivalTime && !stop.departureTime;
};

/**
 * Returns if the station has already been passed by the vehicule.
 * @param {Object} stop Station information.
 */
const isPassed = (stop, time) => {
  // Sometimes stop.departureDelay is undefined.
  const timeToCompare = stop.departureTime || stop.arrivalTime || 0;
  const delayToCompare = stop.departureDelay || stop.arrivalDelay || 0;
  return !isNotStop(stop) && timeToCompare + delayToCompare <= time;
};

const getStationImg = (index, length) => {
  let src = station;
  if (index === 0) {
    src = firstStation;
  } else if (index === length - 1) {
    src = lastStation;
  }
  return src;
};

/**
 * Returns an image for first, middle or last stations.
 * @param {Number} index Index of the station in the list.
 * @param {Number} length Length of the stations list.
 */
const defaultRenderStationImg = (stations, index, length) => {
  const src = getStationImg(index, length);
  return <img src={src} alt="routeScheduleLine" className="rt-route-icon" />;
};

const defaultRenderStation = ({
  lineInfos,
  onStationClick,
  trackerLayer,
  renderStationImg,
  stop,
  idx,
}) => {
  const {
    stationId,
    arrivalDelay,
    departureDelay,
    arrivalTime,
    departureTime,
    cancelled,
    stationName,
  } = stop;
  const { stations } = lineInfos;
  const isFirstStation = idx === 0;
  const isLastStation = idx === stations.length - 1;
  const isStationPassed = isPassed(stop, trackerLayer.currTime);
  const isNotStation = isNotStop(stop);
  return (
    <div
      key={stationId}
      role="button"
      className={[
        'rt-route-station',
        isStationPassed ? ' rt-passed' : '',
        isNotStation ? ' rt-no-stop' : '',
      ].join('')}
      onClick={(e) => onStationClick(stop, e)}
      tabIndex={0}
      onKeyPress={(e) => e.which === 13 && onStationClick(stop, e)}
    >
      <div className="rt-route-delay">
        {typeof arrivalDelay === 'undefined' || isFirstStation || cancelled ? (
          ''
        ) : (
          <span
            className={`rt-route-delay-arrival${` ${getDelayColor(
              arrivalDelay,
            )}`}`}
          >
            {`+${getDelayString(arrivalDelay)}`}
          </span>
        )}
        {typeof departureDelay === 'undefined' || isLastStation || cancelled ? (
          ''
        ) : (
          <span
            className={`rt-route-delay-departure${` ${getDelayColor(
              departureDelay,
            )}`}`}
          >
            {`+${getDelayString(departureDelay)}`}
          </span>
        )}
      </div>
      <div className="rt-route-times">
        <span
          className={`rt-route-time-arrival ${
            cancelled ? 'rt-route-cancelled' : ''
          }`}
        >
          {getHoursAndMinutes(arrivalTime)}
        </span>
        <span
          className={`rt-route-time-departure ${
            cancelled ? 'rt-route-cancelled' : ''
          }`}
        >
          {getHoursAndMinutes(departureTime)}
        </span>
      </div>
      {renderStationImg(stations, idx, stations.length, isNotStation)}
      <div className={cancelled ? 'rt-route-cancelled' : ''}>{stationName}</div>
    </div>
  );
};

const renderRouteIdentifier = ({ routeIdentifier, longName }) => {
  // first part of the id, without leading zeros.
  const id = parseInt(routeIdentifier.split('.')[0], 10);
  if (!longName.includes(id)) {
    return ` (${id})`;
  }
  return null;
};

const defaultRenderHeader = ({ lineInfos, renderHeaderButtons }) => {
  const {
    vehicleType,
    shortName,
    longName,
    color,
    backgroundColor,
    destination,
    routeIdentifier,
  } = lineInfos;
  return (
    <div className="rt-route-header">
      <span
        className="rt-route-icon"
        style={{
          backgroundColor: backgroundColor || bgColors[vehicleType],
          color: color || 'black',
        }}
      >
        {shortName}
      </span>
      <div className="rt-route-title">
        <span className="rt-route-name">{destination}</span>
        <span>
          {longName}
          {renderRouteIdentifier(lineInfos)}
        </span>
      </div>
      <div className="rt-route-buttons">
        {renderHeaderButtons(routeIdentifier)}
      </div>
    </div>
  );
};

const defaultRenderFooter = (props) => {
  const { lineInfos, renderCopyright } = props;
  if (!lineInfos.operator && !lineInfos.publisher) {
    return null;
  }
  return <div className="rt-route-footer">{renderCopyright({ ...props })}</div>;
};

const defaultRenderLink = (text, url) => {
  return (
    <div className="rt-route-copyright-link">
      {url ? (
        <a href={url} target="_blank" rel="noreferrer">
          {text}
        </a>
      ) : (
        <>{text}</>
      )}
    </div>
  );
};

const defaultRenderCopyright = ({ lineInfos }) => {
  return (
    <span className="rt-route-copyright">
      {lineInfos.operator &&
        defaultRenderLink(lineInfos.operator, lineInfos.operatorUrl)}
      {lineInfos.operator && lineInfos.publisher && <span>&nbsp;-&nbsp;</span>}
      {lineInfos.publisher &&
        defaultRenderLink(lineInfos.publisher, lineInfos.publisherUrl)}
      {lineInfos.license && <span>&nbsp;(</span>}
      {lineInfos.license &&
        defaultRenderLink(lineInfos.license, lineInfos.licenseUrl)}
      {lineInfos.license && ')'}
    </span>
  );
};

const propTypes = {
  /**
   * CSS class of the route schedule wrapper.
   */
  className: PropTypes.string,

  /**
   * Trajectory stations informations.
   */
  lineInfos: ReactTransitPropTypes.lineInfos,

  /**
   * Trackerlayer.
   */
  trackerLayer: PropTypes.instanceOf(TrackerLayer).isRequired,

  /**
   * Render Header of the route scheduler.
   */
  renderHeader: PropTypes.func,

  /**
   * Render Footer of the route scheduler.
   */
  renderFooter: PropTypes.func,

  /**
   * Render Copyright of the route scheduler.
   */
  renderCopyright: PropTypes.func,

  /**
   * Render the status of the station image.
   */
  renderStationImg: PropTypes.func,

  /**
   * Render a station.
   */
  renderStation: PropTypes.func,

  /**
   * Function triggered on station's click event.
   */
  onStationClick: PropTypes.func,

  /**
   * Function to render header buttons.
   */
  renderHeaderButtons: PropTypes.func,
};

const defaultProps = {
  className: 'rt-route-schedule',
  lineInfos: null,
  renderHeader: defaultRenderHeader,
  renderStation: defaultRenderStation,
  renderStationImg: defaultRenderStationImg,
  renderCopyright: defaultRenderCopyright,
  renderFooter: defaultRenderFooter,
  renderHeaderButtons: () => null,
  onStationClick: () => {},
};

/**
 * RouteSchedule displays information, stops and punctuality about the clicked route.
 */
function RouteSchedule(props) {
  const {
    lineInfos,
    className,
    renderStation,
    renderHeader,
    renderFooter,
  } = props;

  if (!lineInfos) {
    return null;
  }

  return (
    <div className={className}>
      {renderHeader({ ...props })}
      <div className="rt-route-body">
        {lineInfos.stations.map((stop, idx) =>
          renderStation({ ...props, stop, idx }),
        )}
      </div>
      {renderFooter({ ...props })}
    </div>
  );
}

RouteSchedule.propTypes = propTypes;
RouteSchedule.defaultProps = defaultProps;

export default React.memo(RouteSchedule);
