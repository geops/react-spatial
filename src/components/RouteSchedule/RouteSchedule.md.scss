.rt-route-schedule-example {
  font-family: $font-family;
  position: relative;
  display: flex;

  .rt-route-schedule {
    display: flex;
    flex-direction: column;
    height: 300px;
    width: 50%;

    .rt-route-header {
      padding-top: 0;
    }

    .rt-route-footer {
      padding-bottom: 0;
    }

    .rt-route-header,
    .rt-route-footer {
      flex: 0 0 auto; // 'auto' is for IE11
    }

    .rt-route-body {
      overflow-y: auto;
      flex: 2;
    }
  }

  .rs-map {
    flex: 1;
  }
}
