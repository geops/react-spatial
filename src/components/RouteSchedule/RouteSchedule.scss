.rt-route-schedule {
  background-color: white;
  width: 350px;
  overflow: hidden;

  .rt-route-header {
    display: flex;
    align-items: center;
    padding: 15px 10px 0 10px;

    .rt-route-title {
      display: flex;
      flex-direction: column;

      .rt-route-name {
        padding-bottom: 8px;
        font-weight: bold;
      }
    }

    .rt-route-buttons {
      margin-left: auto;
      display: flex;
    }

    .rt-route-icon {
      border-radius: 20px;
      min-width: 20px;
      height: 20px;
      line-height: 20px;
      border: solid black 2px;
      padding: 5px;
      display: block;
      float: left;
      margin: 15px;
      margin-top: 10px;
      text-align: center;
      font-size: 14px;
      font-weight: bold;
    }
  }

  .rt-route-footer {
    padding: 20px;
    display: flex;
    align-items: center;
  }

  .rt-route-copyright {
    display: flex;
    flex-wrap: wrap;
  }

  .rt-route-body {
    font-size: 14px;
    padding: 0 20px;

    .rt-route-station {
      display: flex;
      align-items: center;
      cursor: pointer;
      border-radius: 4px;

      &:first-child,
      &:last-child {
        font-weight: bold;
      }

      &:hover {
        color: white;
        background-color: $brand-secondary;
      }

      .rt-route-times,
      .rt-route-delay {
        display: flex;
        flex-direction: column;
        width: 40px;
        min-width: 40px;
        padding: 0 3px;

        .green {
          color: rgb(0, 160, 12);
        }

        .orange {
          color: rgb(247, 191, 0);
        }

        .light-red {
          color: rgb(255, 74, 0);
        }

        .middle-red {
          color: rgb(232, 0, 0);
        }

        .dark-red {
          color: rgb(237, 0, 76);
        }
      }
    }

    .rt-route-station.rt-passed,
    .rt-route-station.rt-no-stop {
      opacity: 0.7;

      .rt-route-delay {
        span {
          display: none;
        }
      }
    }

    .rt-route-station.rt-passed {
      .rt-route-icon-mask {
        height: 0;
      }
    }

    .rt-route-station:first-child {
      .rt-route-time-arrival {
        display: none;
      }
    }

    .rt-route-station:last-child {
      .rt-route-time-departure {
        display: none;
      }
    }
  }

  .rt-route-cancelled {
    text-decoration: line-through;
    color: rgb(236, 43, 43);
  }
}
