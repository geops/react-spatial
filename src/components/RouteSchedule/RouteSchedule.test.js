import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-date-mock';
import { TrajservLayer } from 'mobility-toolbox-js/ol';
import RouteSchedule from '.';

configure({ adapter: new Adapter() });
const RealDate = Date;

const lineInfos = {
  backgroundColor: 'ff8a00',
  destination: 'Station name',
  id: 9959310,
  routeIdentifier: '03634.003849.004:9',
  longName: 'T 3',
  shortName: '3',
  operator: 'foo',
  operatorUrl: 'foo.ch',
  publisher: 'bar',
  publisherUrl: 'bar.ch',
  stations: [
    {
      stationId: '1',
      stationName: 'first stop',
      coordinates: [8.51772, 47.3586],
      arrivalDelay: 60000, // +1m
      arrivalTime: 1571729580000,
      departureDelay: 60000,
      departureTime: 1571729580000,
    },
    {
      stationId: '2',
      stationName: 'second stop',
      coordinates: [8.54119, 47.36646],
      arrivalDelay: 0, // +0
      arrivalTime: 1571729903000,
      departureDelay: 120000, // +2m
      departureTime: 1571729903000,
    },
    {
      stationId: '4',
      stationName: 'no stop',
      coordinates: [8.54119, 47.36646],
      arrivalDelay: null, // +0
      arrivalTime: 0,
      departureDelay: null, // +2m
      departureTime: 0,
    },
    {
      stationId: '3',
      stationName: 'third stop',
      coordinates: [8.54119, 50],
      arrivalDelay: 240000, // +4m
      arrivalTime: 1571730323000,
      departureDelay: 0, // +0
      departureTime: 0,
    },
  ],
  vehicleType: 0,
};

describe('RouteSchedule', () => {
  beforeEach(() => {
    global.Date = jest.fn(() => ({
      getHours: () => {
        return 9;
      },
      getMinutes: () => {
        return 1;
      },
    }));
    Object.assign(Date, RealDate);
  });

  afterEach(() => {
    global.Date = RealDate;
  });

  test('matches snapshots.', () => {
    const trackerLayer = new TrajservLayer();
    const component = renderer.create(
      <RouteSchedule
        lineInfos={lineInfos}
        trackerLayer={trackerLayer}
        setCenter={() => {}}
        renderHeaderButtons={() => <div>Button</div>}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // to test: on station click
  // to test: time formating
  // to test: delay formating
  // to test: delay color
  // to test: no arrival delay on first station
  // to test: no arrival date on first station
  // to test: no departure delay on last station
  // to test: no departure date on last station
  // to test: font bold on first and last station
});
