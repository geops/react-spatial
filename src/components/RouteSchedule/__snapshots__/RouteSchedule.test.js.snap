// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteSchedule matches snapshots. 1`] = `
<div
  className="rt-route-schedule"
>
  <div
    className="rt-route-header"
  >
    <span
      className="rt-route-icon"
      style={
        Object {
          "backgroundColor": "ff8a00",
          "color": "black",
        }
      }
    >
      3
    </span>
    <div
      className="rt-route-title"
    >
      <span
        className="rt-route-name"
      >
        Station name
      </span>
      <span>
        T 3
         (3634)
      </span>
    </div>
    <div
      className="rt-route-buttons"
    >
      <div>
        Button
      </div>
    </div>
  </div>
  <div
    className="rt-route-body"
  >
    <div
      className="rt-route-station"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="rt-route-delay"
      >
        <span
          className="rt-route-delay-departure green"
        >
          +1m
        </span>
      </div>
      <div
        className="rt-route-times"
      >
        <span
          className="rt-route-time-arrival"
        >
          09:01
        </span>
        <span
          className="rt-route-time-departure"
        >
          09:01
        </span>
      </div>
      <img
        alt="routeScheduleLine"
        className="rt-route-icon"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAyCAYAAABcfPsmAAAABmJLR0QA/wD/AP+gvaeTAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH3gYeCBQzRAvlCAAAABl0RVh0Q29tbWVudABDcmVhdGVkIHdpdGggR0lNUFeBDhcAAAD2SURBVEjH7ZPNEYIwEEafPwXQgSnBBhydoRDog4N44GAVQiMODBXQAcPVCx3owUUzmjAyenO/S0J280h2v4BKpVKp/lQzb2STGcAAW2u1AlrqpJ0GvMP2AtxZkRJogYMPuhiBxQK0ZYA1ELAKG7pz/7p96fjJTmDIaSoZh+sbiVdA/rp57gBG1rwAUuokBVL5duWNAu2a5Y9a3cfckzcK/EouYGnNY2nS0KzYkzfalMK6TgQYNlnr8GTxKbCUWg22MY6c3HfCdx92555V2ACBeM4F8xrb15QLcPLEjtOf3vPVXN/W6mQ2tcs/t40CFahABSrwB7oBXXk7PEfz+n8AAAAASUVORK5CYII="
      />
      <div
        className={null}
      >
        first stop
      </div>
    </div>
    <div
      className="rt-route-station"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="rt-route-delay"
      >
        <span
          className="rt-route-delay-arrival green"
        >
          +0
        </span>
        <span
          className="rt-route-delay-departure green"
        >
          +2m
        </span>
      </div>
      <div
        className="rt-route-times"
      >
        <span
          className="rt-route-time-arrival"
        >
          09:01
        </span>
        <span
          className="rt-route-time-departure"
        >
          09:01
        </span>
      </div>
      <img
        alt="routeScheduleLine"
        className="rt-route-icon"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAyCAYAAABcfPsmAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH3gcKCgsxhsC3rAAAABl0RVh0Q29tbWVudABDcmVhdGVkIHdpdGggR0lNUFeBDhcAAADoSURBVEjH7ZbBDYJAEEUfagF0ICXQAJGEQtw+OIgHDlYhNEIwVGAHxKsXOtCDG9wIu1kTowdnLjPMzrwMOz+BAJcl5W2S6/LA1bLgwyZAAQpQgAL8OjApo7fy2paOph0Qz5yGrLMzl2bwAz5hyjJE7IKuZhpSA9YDJ+0jYKO90vnK5w63RlwDBV1eAIV+nqtzAlMjrujyXn+P+5eJ0p/JpjViNcrk4ZWlzrmU2nidLRCRlOZSzDovYKvvSmnInJAr24RTHV6agXV2BkKLsCtgPy7LcylX4Gg5O9hgAIH8zglQgAIU4L8C709SOkzp8yUKAAAAAElFTkSuQmCC"
      />
      <div
        className={null}
      >
        second stop
      </div>
    </div>
    <div
      className="rt-route-station rt-no-stop"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="rt-route-delay"
      >
        <span
          className="rt-route-delay-arrival green"
        >
          +0
        </span>
        <span
          className="rt-route-delay-departure green"
        >
          +0
        </span>
      </div>
      <div
        className="rt-route-times"
      >
        <span
          className="rt-route-time-arrival"
        >
          
        </span>
        <span
          className="rt-route-time-departure"
        >
          
        </span>
      </div>
      <img
        alt="routeScheduleLine"
        className="rt-route-icon"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAyCAYAAABcfPsmAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH3gcKCgsxhsC3rAAAABl0RVh0Q29tbWVudABDcmVhdGVkIHdpdGggR0lNUFeBDhcAAADoSURBVEjH7ZbBDYJAEEUfagF0ICXQAJGEQtw+OIgHDlYhNEIwVGAHxKsXOtCDG9wIu1kTowdnLjPMzrwMOz+BAJcl5W2S6/LA1bLgwyZAAQpQgAL8OjApo7fy2paOph0Qz5yGrLMzl2bwAz5hyjJE7IKuZhpSA9YDJ+0jYKO90vnK5w63RlwDBV1eAIV+nqtzAlMjrujyXn+P+5eJ0p/JpjViNcrk4ZWlzrmU2nidLRCRlOZSzDovYKvvSmnInJAr24RTHV6agXV2BkKLsCtgPy7LcylX4Gg5O9hgAIH8zglQgAIU4L8C709SOkzp8yUKAAAAAElFTkSuQmCC"
      />
      <div
        className={null}
      >
        no stop
      </div>
    </div>
    <div
      className="rt-route-station"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="rt-route-delay"
      >
        <span
          className="rt-route-delay-arrival orange"
        >
          +4m
        </span>
      </div>
      <div
        className="rt-route-times"
      >
        <span
          className="rt-route-time-arrival"
        >
          09:01
        </span>
        <span
          className="rt-route-time-departure"
        >
          
        </span>
      </div>
      <img
        alt="routeScheduleLine"
        className="rt-route-icon"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAyCAYAAABcfPsmAAAABmJLR0QA/wD/AP+gvaeTAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH3gYeCBMwkkMidQAAABl0RVh0Q29tbWVudABDcmVhdGVkIHdpdGggR0lNUFeBDhcAAAEeSURBVEjH7ZSxbcMwEEWflAzgJrVZpswCRgR4ECtTpGARpfAe8iKGDNYpXKUVPIGa1E5h2iBkHkMCApLA/IAgSvfv607/yIIQFuvj1Tuji1BKycSYXLAItAtwFHOMTqjwJDZRy7FiAu/vm5IF/4WgsANieWUKOSZeJif98LEbdDl0Hj4Cn57IM/CB0V8p56ECXoVPvQAPUh13gtgbUAs5T8CM+XLPYTuMg/eehMoR64GdvSvbrrLxHdDGtLxy1hugwegGaOyzjxcUrJx1i9G9Heh+VFH1a3PYOevamnQ2qxZ4QVM2TjsrQLFYu6a4vCjBzv6r2oooD6eVKryew8N2YL7cAzNgGAl29nq/mJWw9ZSnzdNMCmIZGRkZGTeNb4FoRhP3GY83AAAAAElFTkSuQmCC"
      />
      <div
        className={null}
      >
        third stop
      </div>
    </div>
  </div>
  <div
    className="rt-route-footer"
  >
    <span
      className="rt-route-copyright"
    >
      <div
        className="rt-route-copyright-link"
      >
        <a
          href="foo.ch"
          rel="noreferrer"
          target="_blank"
        >
          foo
        </a>
      </div>
      <span>
         - 
      </span>
      <div
        className="rt-route-copyright-link"
      >
        <a
          href="bar.ch"
          rel="noreferrer"
          target="_blank"
        >
          bar
        </a>
      </div>
    </span>
  </div>
</div>
`;
